src/components/Attachments/AttachmentsActions.js

```diff
@@ -0,0 +1,49 @@
+import { createAction } from 'redux-actions';
+
+export const addUploadingFiles = createAction(
+  'ADD_UPLOADING_FILES',
+  (files, uploadFolderId) => ({
+    files,
+    uploadFolderId,
+  }),
+);
+
+export const fileUploadInitialized = createAction(
+  'FILE_UPLOAD_INITIALIZED',
+  (file, uploadFolderId) => ({ file, uploadFolderId }),
+);
+
+export const fileUploadGetLocation = createAction(
+  'FILE_UPLOAD_GET_LOCATION',
+  (file, uploadFolderId) => ({ file, uploadFolderId }),
+);
+
+export const fileUploadingStarted = createAction(
+  'FILE_UPLOADING_STARTED',
+  (file, uploadFolderId) => ({
+    file,
+    uploadFolderId,
+  }),
+);
+
+export const fileUploadingProgressUpdated = createAction(
+  'FILE_UPLOADING_PROGRESSS_UPDATED',
+  (file, uploadFolderId, progress) => ({ file, uploadFolderId, progress }),
+);
+
+export const fileUploadingSuccess = createAction(
+  'FILE_UPLOADING_SUCCESS',
+  (file, uploadFolderId, result) => ({ file, uploadFolderId, result }),
+);
+
+export const fileUploadingFailure = createAction(
+  'FILE_UPLOADING_FAILURE',
+  (file, uploadFolderId, error) => ({ file, uploadFolderId, error }),
+);
+
+export const retryFileUpload = createAction(
+  'RETRY_FILE_UPLOAD',
+  (file, uploadFolderId) => ({ file, uploadFolderId }),
```

Comment:
Why do we need `(file, uploadFolderId)` function? Why not just use `({ file, uploadFolderId })` everywhere?

Reply:
The same question for all actions in this file.

Reply:
popular practice to design action creators as functions

for functions, arguments are a more natural way to pass data

you can invoke action creator in next way:

```
const action = retryFileUpload(file1, folder1);
```

instead of:
```
const action = retryFileUpload({
  file: file1,
  uploadFolderId: folder1
});
```

if you decide to change the structure of action payload you don't need to change all code where you invoke your action creator as a function

additionally, it's an encapsulation of logic of creating action payload

instead of duplication a lot of code where you are making an object to pass as an argument to the action creator you are making it only in one place